app/uk/gov/hmrc/securemessage/models/core/CustomerEnrolment.scala

```diff
@@ -30,8 +30,11 @@ object CustomerEnrolment {
     Json.reads[CustomerEnrolment]
   }
 
-  def parse(enrolmentString: String): CustomerEnrolment = {
+  def parse(enrolmentString: String): Either[String, CustomerEnrolment] = {
```

Comment:
i would have the left as a specialised error.

Reply:
This is now actually the correct behaviour. It now returns a Left with the correct message if the CustomerEnrolment cannot be parsed. Before it would throw an exception.

Reply:
yeah, an i'm saying instead of String, have a specialised Error for the left, similar to https://github.com/hmrc/secure-message/blob/master/app/uk/gov/hmrc/securemessage/SecureMessageError.scala#L51
And then handle it in the ErrorHandling: https://github.com/hmrc/secure-message/blob/master/app/uk/gov/hmrc/securemessage/controllers/ErrorHandling.scala#L41